src/WLR/Types/Output.hsc

```diff
@@ -0,0 +1,431 @@
+{-# LANGUAGE EmptyDataDeriving, PatternSynonyms #-}
+
+module WLR.Types.Output where
+
+#define WLR_USE_UNSTABLE
+#include <wlr/types/wlr_output.h>
+
+import Foreign
+import Foreign.C.Types
+
+import PIXMAN.Pixman
+import WL.ServerCore
+import WL.ServerProtocol
+import WL.Utils
+import WLR.Backend
+import WLR.Render.Allocator
+import WLR.Render.DrmFormatSet
+import WLR.Render.Interface
+import WLR.Render.Renderer
+import WLR.Render.Swapchain
+import WLR.Render.Texture
+import WLR.Types.Buffer
+import WLR.Types.OutputLayer
+import WLR.Util.Addon
+import WLR.Util.Box
+
+-- A struct from <time.h>. I think this should only be handled by
+-- standard library code, so an opaque pointer should suffice.
+type TIMESPEC = ()
+
+{{ enum
+    WLR_output_mode_aspect_ratio,
+    WLR_OUTPUT_MODE_ASPECT_RATIO_NONE,
+    WLR_OUTPUT_MODE_ASPECT_RATIO_4_3,
+    WLR_OUTPUT_MODE_ASPECT_RATIO_16_9,
+    WLR_OUTPUT_MODE_ASPECT_RATIO_64_27,
+    WLR_OUTPUT_MODE_ASPECT_RATIO_256_135
+}}
+
+{{ struct
+    wlr/types/wlr_output.h,
+    wlr_output_mode,
+    width,                CInt,
+    height,               CInt,
+    refresh,              CInt,
+    preferred,            CBool,
+    picture_aspect_ratio, WLR_output_mode_aspect_ratio,
+    link,                 WL_list
+}}
+
+{{ struct
+    wlr/types/wlr_output.h,
+    wlr_output_cursor,
+    output,      Ptr WLR_output,
+    x,           CDouble,
+    y,           CDouble,
+    enabled,     CBool,
+    visible,     CBool,
+    width,       CUInt,
+    height,      CUInt,
```

Comment:
Similarly to previous comment, these should be `Word32`.

(There are many more instances of this throughout the file, but I’ll refrain from adding a comment on every single one.)